api/cpp/include/slint.h

```diff
@@ -347,6 +356,18 @@ class Window
         return inner.set_rendering_notifier(std::forward<F>(callback));
     }
 
+    /// This function allows registering a callback that's invoked when the user tries to close
+    /// a window.
+    /// The callback has to return a CloseRequestResponse.
+    template<typename F>
+    void on_close_requested(F &&callback) const
+    {
+        static_assert(std::is_invocable_v<Functor>, "Functor callback must be callable");
+        static_assert(std::is_same_v<std::invoke_result_t<F>, CloseRequestResponse>,
```

Comment:
ideally this would be a concept.
Note that this force the type to be exactly CloseRequestResponse and not something that can be implictly converted to it, but i guess that's alright.